addons/point_of_sale/models/pos_session.py

```diff
@@ -2102,6 +2108,12 @@ def load_order_fiscal_positions(self, fp_ids):
         fiscal_position_tax = self.env['account.fiscal.position.tax'].search_read(**tax_params['search_params'])
         fiscal_position_by_id = {fpt['id']: fpt for fpt in fiscal_position_tax}
         for fiscal_position in fps:
+            if not self.config_id.tax_regime_selection:
+                self.config_id.tax_regime_selection = True
+            fiscal_position_id = self.env['account.fiscal.position'].browse(fiscal_position['id'])
+            self.config_id.fiscal_position_ids += fiscal_position_id
+            if not self.config_id.default_fiscal_position_id:
+                self.config_id.default_fiscal_position_id = fiscal_position_id
```

Comment:
Same here, it's not because we allow a fiscal position that we want is as the default one